plugin-offline/src/main/java/com/mapbox/mapboxsdk/plugins/offline/ui/RegionSelectionFragment.java

```diff
@@ -0,0 +1,186 @@
+package com.mapbox.mapboxsdk.plugins.offline.ui;
+
+import android.graphics.PointF;
+import android.graphics.RectF;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.geometry.LatLngBounds;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.offline.OfflineTilePyramidRegionDefinition;
+import com.mapbox.mapboxsdk.plugins.offline.R;
+import com.mapbox.mapboxsdk.style.sources.VectorSource;
+import com.mapbox.services.commons.geojson.Feature;
+
+import java.util.List;
+
+import timber.log.Timber;
+
+public class RegionSelectionFragment extends Fragment implements OnMapReadyCallback,
+  MapboxMap.OnCameraIdleListener {
+
+  public static final String TAG = "OfflineRegionSelectionFragment";
+  private static final String[] LAYER_IDS = new String[] {
+    "place-city-lg-n", "place-city-lg-s", "place-city-md-n", "place-city-md-s", "place-city-sm"
+  };
+  private static final String[] SOURCE_LAYER_IDS = new String[] {
+    "place_label", "state_label", "country_label"
+  };
+
+  private RegionSelectedCallback selectedCallback;
+  private TextView regionNameTextView;
+  private MapboxMap mapboxMap;
+  private RectF boundingBox;
+  private MapView mapView;
+  private View rootView;
+  String regionName;
+
+  public static RegionSelectionFragment newInstance() {
+    return new RegionSelectionFragment();
+  }
+
+  @Nullable
+  @Override
+  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                           @Nullable Bundle savedInstanceState) {
+    rootView = inflater.inflate(R.layout.mapbox_offline_region_selection_fragment, container, false);
+    mapView = rootView.findViewById(R.id.mapbox_offline_region_selection_map_view);
+    regionNameTextView = rootView.findViewById(R.id.mapbox_offline_region_name_text_view);
+    return rootView;
+  }
+
+  @Override
+  public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+    bindClickListeners();
+  }
+
+  public RegionSelectedCallback getSelectedCallback() {
+    return selectedCallback;
+  }
+
+  public void setSelectedCallback(@NonNull RegionSelectedCallback selectedCallback) {
```

Comment:
onAttach is exposed on Fragment for setting callbacks, see https://developer.android.com/training/basics/fragments/communicating.html, no need for specific setter

Reply:
Fragment can be directly used by users rather than using the activity if they need to.